syntax = "proto3";

package ytrpchead;

message ReqHead {
  uint32 req_id = 1;  // 请求唯一id

  bytes func = 2;  // 调用的服务接口名，格式【/pkg.service/func】

  uint64 ddl_ms = 3;  // ddl时间戳，单位ms

  map<string, bytes> context_kv = 4;  // 上下文kv对，用于框架层
}

message RspHead {
  uint32 req_id = 1;  // 请求唯一id

  int32 ret_code = 2;  // 框架层面返回码

  int32 func_ret_code = 3;  // 业务层面返回码

  bytes func_ret_msg = 4;  // 业务层面返回信息
}
